sphinx_gallery/scrapers.py

```diff
@@ -204,15 +204,25 @@ def matplotlib_scraper(block, block_vars, gallery_conf, **kwargs):
             )
         )
 
+    # Determine whether single-img should be converted to multi-img
+    convert_to_multi_image = True  # default is to convert
+    if block_vars.get("multi_image") is not None:  # block setting takes precedence
+        convert_to_multi_image = block_vars["multi_image"] != "single"
+    elif block_vars["file_conf"].get("multi_image") is not None:  # then file setting
+        convert_to_multi_image = block_vars["file_conf"]["multi_image"] != "single"
```

Comment:
This in the scraper prioritises checking the block-specific setting, then falls back to the file-wide setting.